src/docker_hub.ml

```diff
@@ -4,14 +4,13 @@
 (* More complete script also available here: https://github.com/moby/moby/blob/924edb948c2731df3b77697a8fcc85da3f6eef57/contrib/download-frozen-image-v2.sh *)
 
 type fetch_errors = [
-  | `Api_error of Http_lwt_client.response * string option
+  | `Api_error of Httpcats.response * string option
   | `Malformed_json of string
-  | `Msg of string
+  | Httpcats.error
```

Comment:
Do users of httpcats really need a list of errors that detailed?

Reply:
I think it's pretty good and it fits in quite well with your code :+1:. A transformation could be `reword_error (msgf '%a' Httpcats.pp_error)` to produce only a value of type `[> Msg of string ]`. But specifying whether the error comes from `h1` or `h2` or the underlying protocol (`tcp` or `tls`) or the occurrence of an unknown exception (often raised by the user (him/her)self) can be useful in more precise cases.